node/src/components/block_executor.rs

```diff
@@ -439,16 +442,18 @@ impl<REv: ReactorEventT> Component<REv> for BlockExecutor {
                 effect_builder
                     .get_block_at_height_local(finalized_block.height())
                     .event(move |maybe_block| {
-                        Event::BlockAlreadyExists(maybe_block.is_some(), finalized_block)
+                        Event::BlockAlreadyExists(maybe_block.map(Box::new), finalized_block)
                     })
             }
-            Event::BlockAlreadyExists(exists, finalized_block) => {
-                if !exists {
+            Event::BlockAlreadyExists(maybe_block, finalized_block) => {
+                if let Some(block) = maybe_block {
+                    effect_builder
+                        .handle_linear_chain_block(block.take_header())
```

Comment:
Instead of calling `handle_linear_chain_block` we should be calling `effect_builder.announce_linear_chain_block`. This will forward the block to the linear chain and it already calls `handle_linear_chain_block`.

Reply:
Although, the code acts on that announcement in ways that we may not want to:
* announces all the deploys as `DeployProcessed`
* announces block as `BlockAdded`
* triest storing the block and its execution effect back again

Reply:
I think that was why we (me and Ed) chose to handle it this way instead of via an announcement. It looked like the announcement was a step too early, we don't want to announce the block as if it was new, we just want consensus to handle it and emit relevant announcements to make synchronization progress.